cmake/CorrosionGenerator.cmake

```diff
@@ -52,7 +52,9 @@ function(_generator_add_package_targets workspace_manifest_path package_manifest
         set(kinds)
         foreach(ix RANGE ${target_kind_len-1})
             string(JSON kind GET "${target_kind}" ${ix})
-            list(APPEND kinds ${kind})
+            if(NOT COR_CRATE_TYPES OR ${kind} IN_LIST COR_CRATE_TYPES)
```

Comment:
`COR_CRATE_TYPES` should be passed as a parameter to this function.
This currently works because in CMake we can also see all variables defined in parent scopes, but we shouldn't rely on this.